src/mixins/ajax.js

```diff
@@ -7,29 +7,13 @@ export default {
      */
     loading: {
       type: Boolean,
-      default: false
+      default: false,
     },
-
-    /**
-     * Accept a callback function that will be
-     * run when the search text changes.
-     *
-     * loading() accepts a boolean value, and can
-     * be used to toggle a loading class from
-     * the onSearch callback.
-     *
-     * @param {search}  String          Current search text
-     * @param {loading} Function(bool)  Toggle loading class
-     */
-    onSearch: {
```

Comment:
Do you think it would make sense to leave this in for now and add a depreciation warning instead?

Reply:
I think that if the depreciation notices are backed filled it will be ok. I was just concerned about people updating without knowing that this was being removed.

Great work consolidating the event structure 👍